wandb/sdk/launch/agent/agent.py

```diff
@@ -131,7 +131,7 @@ def run_job(self, job: Dict[str, Any]) -> None:
         resource = launch_spec.get("resource") or "local"
         backend_config: Dict[str, Any] = {
             PROJECT_DOCKER_ARGS: {},
-            PROJECT_SYNCHRONOUS: True,
+            PROJECT_SYNCHRONOUS: False,  # agent always runs async
```

Comment:
Let's drop this constant and remove references to it

Reply:
regular `wandb launch` currently has some flexibility based on this constant to support both sync and async modes (local run always runs as synchronous). do we wanna keep that or just force local to be sync all the time

Reply:
Good point, let's leave this for now